### PR TITLE
KAFKA-16828: RackAwareTaskAssignorTest failed

### PR DESCRIPTION
**Pull request by brandboat**
related to https://issues.apache.org/jira/browse/KAFKA-16828

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ private boolean validateTopicPartitionRack(final boolean changelogTopics) {
      *
      * @return whether the operation successfully completed and the rack information is valid.
      */
-    public static boolean validateTopicPartitionRack(final Cluster fullMetadata,
+    public boolean validateTopicPartitionRack(final Cluster fullMetadata,
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java]: If this method is not static method, we don't need to keep two varieties. Could you please merge two `validateTopicPartitionRack` into single one. Also, the new method can be package-private

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ public boolean populateTopicsToDescribe(final Set<String> topicsToDescribe,
         return populateTopicsToDescribe(fullMetadata, changelogPartitionsForTask, partitionsForTask, changelog, topicsToDescribe, racksForPartition);
     }
 
-    public static boolean populateTopicsToDescribe(final Cluster fullMetadata,
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java:142]: ditto.

**Review by chia7712 — COMMENTED** [review]
@brandboat thanks for this hot-fix!

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java (hunk 2 above)
Comment (chia7712) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java:142]: I feel addressing this comment can fix all failed tests.

Comment (brandboat) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java:142]: Thanks, the solution is cleaner.

**Pull request merged**